### PR TITLE
Add namespace lister

### PR DESCRIPTION
**Pull request by LenaHoinkis**
### Description

&lt;Add detailed description for reviewers here.&gt;

### Submitter checklist

- [ ] Change has been tested (on a back-end cluster)
- [ ] (If applicable) Jira backlog item ID added to the PR title and the [changelog.yaml] entry
- [ ] [changelog.yaml] entry with upgrade notes is prepared and appropriate for the audience affected by the change (users or developer, depending on the change)
- [ ] Semantic version diffed against [last release][releases] and updated accordingly. In this project the version has to be maintained here:
    - [/charts/steward/Chart.yaml](https://github.com/SAP/stewardci-core/blob/master/charts/steward/Chart.yaml) (`version` and `appVersion`)

In case dependencies have been updated:
- [ ] Links to external changelogs, since the last release of our component, added to the [changelog.yaml] entry (description).
- [ ] Changelogs read thoroughly, potential impact described, upgrade notes prepared (if necessary)
- [ ] Check if dependency updates affect our semantic version increment.

### Reviewer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There is at least 1 approval for the pull request and no outstanding requests for change
- [ ] All voter checks have passed
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] The Pull Request title is understandable and reflects the changes well
- [ ] The Pull Request description is understandable and well documented
- [ ] [changelog.yaml] entry for this Pull Request has been added
    - [ ] Changelog entry contains all required information
    - [ ] 'Upgrade notes' are documented in changelog.yaml (if required)

[changelog.yaml]: https://github.com/SAP/stewardci-core/changelog.yaml
[releases]: https://github.com/SAP/stewardci-core/releases

### CODE REVIEW

File: pkg/tenantctl/controller_test.go [hunk 1]
```diff
@@ -124,7 +124,10 @@ func Test_Controller_syncHandler_AddsFinalizer(t *testing.T) {
 		// the tenant
 		fake.Tenant(tenantID, clientNSName),
 	)
-	ctl := NewController(cf, NewMetrics())
+
+	stopCh, ctl := startController(t, cf, false)
+	defer stopController(t, stopCh)
```
Comment (anbrsap) [inline pkg/tenantctl/controller_test.go:129]: I don't like that the controller gets started for those tests where we explicitly call e.g. `syncHandler()`.
Instead, just _create_ the controller as before and start `cf.CoreV1InformerFactory()` in addition.

You may use a helper function that returns a deferrable stop function:

```go
ctl := NewController(cf, NewMetrics())
defer startCoreV1Informer(ctl)()
```

```go
func startCoreV1Informer(ctl *Controller) func() {
	stopCh := make(chan struct{}, 0)
	ctl.factory.CoreV1InformerFactory().Start(stopCh)
	return func() {
		stopCh <- struct{}{}
	}
}
```

File: pkg/k8s/fake/clientFactory.go [hunk 2]
```diff
@@ -40,9 +41,11 @@ func NewClientFactory(objects ...runtime.Object) *ClientFactory {
 	stewardInformerFactory := stewardinformer.NewSharedInformerFactory(stewardClientset, time.Minute*10)
 	tektonClientset := tektonclientfake.NewSimpleClientset(tektonObjects...)
 	tektonInformerFactory := tektoninformers.NewSharedInformerFactory(tektonClientset, time.Minute*10)
+	kubernetesClientset := kubernetes.NewSimpleClientset(kubernetesObjects...)
+	//kubernetesInformerFactory := informers.NewSharedInformerFactory(kubernetesClientset, time.Minute*10)
```
Comment (rinckm) [inline pkg/k8s/fake/clientFactory.go:45]:
```suggestion
```

File: pkg/k8s/fake/clientFactory.go [hunk 3]
```diff
@@ -93,6 +96,10 @@ func (f *ClientFactory) KubernetesClientset() *kubernetes.Clientset {
 	return f.kubernetesClientset
 }
 
+func (f *ClientFactory) KubernetesInterface() kubernetesTrue.Interface {
```
Comment (rinckm) [inline pkg/k8s/fake/clientFactory.go:99]: Please add docu

File: pkg/k8s/fake/clientFactory.go [hunk 4]
```diff
@@ -15,6 +15,7 @@ import (
 	runtime "k8s.io/apimachinery/pkg/runtime"
 	dynamic "k8s.io/client-go/dynamic"
 	dynamicfake "k8s.io/client-go/dynamic/fake"
+	kubernetesTrue "k8s.io/client-go/kubernetes"
 	kubernetes "k8s.io/client-go/kubernetes/fake"
```
Comment (rinckm) [inline pkg/k8s/fake/clientFactory.go:19]: Maybe you should rename to kubernetes and kubernetesfake

File: pkg/k8s/fake/clientFactory.go [hunk 5]
```diff
@@ -93,6 +95,11 @@ func (f *ClientFactory) KubernetesClientset() *kubernetes.Clientset {
 	return f.kubernetesClientset
 }
 
+// KubernetesClientset returns an interface of the Kubernetes fake clientset.
```
Comment (rinckm) [inline pkg/k8s/fake/clientFactory.go:98]:
```suggestion
// KubernetesInterface returns an interface of the Kubernetes fake clientset.
```

File: pkg/k8s/clientFactory.go [hunk 6]
```diff
@@ -23,6 +23,9 @@ type ClientFactory interface {
 	// CoreV1 returns the core/v1 Kubernetes client
 	CoreV1() corev1.CoreV1Interface
 
+	// KubernetesClientset returns Kubernetes clientset
+	KubernetesInterface() kubernetes.Interface
```
Comment (anbrsap) [inline pkg/k8s/clientFactory.go:27]: I would use the name `KubernetesClientSet()` instead, as package `k8s.io/client-go/kubernetes` provides the Kubernetes client set, and `Interface` is just a _local_ name from this package.

File: pkg/tenantctl/controller.go [hunk 7]
```diff
@@ -90,8 +107,9 @@ func (c *Controller) getNamespaceManager(config clientConfig) k8s.NamespaceManag
 func (c *Controller) Run(threadiness int, stopCh <-chan struct{}) error {
 	defer utilruntime.HandleCrash()
 	defer c.workqueue.ShutDown()
+	c.namespaceFactory.Start(stopCh)
```
Comment (jia-jerry) [inline pkg/tenantctl/controller.go:110]: To be consistent, please puth the informerfactor `Start` with https://github.com/SAP/stewardci-core/blob/master/cmd/run_controller/main.go#L78

**Review by jia-jerry — CHANGES_REQUESTED** [review]

**Pull request closed without merging**